modules/richaudienceBidAdapter.js

```diff
@@ -138,3 +127,42 @@ export const spec = {
 };
 
 registerBidder(spec);
+
+function raiGetSizes(bid) {
+  let raiNewSizes;
+  if (bid.mediaTypes && bid.mediaTypes.banner && bid.mediaTypes.banner.sizes) {
+    raiNewSizes = bid.mediaTypes.banner.sizes
+  } else {
+    raiNewSizes = bid.sizes
+  }
+  if (raiNewSizes != null) {
+    return raiNewSizes.map(size => ({
+      w: size[0],
+      h: size[1]
+    }));
+  }
+}
+
+function raiSetEids(bid) {
+  let eids = [];
+
+  if (bid && bid.userId) {
+    raiSetUserId(bid, eids, 'id5-sync.com', utils.deepAccess(bid, `userId.id5id`));
+    raiSetUserId(bid, eids, 'pubcommon', utils.deepAccess(bid, `userId.pubcid`));
+    raiSetUserId(bid, eids, 'criteo.com', utils.deepAccess(bid, `userId.criteoId`));
+    raiSetUserId(bid, eids, 'liveramp.com', utils.deepAccess(bid, `userId.idl_env`));
+    raiSetUserId(bid, eids, 'liveintent.com', utils.deepAccess(bid, `userId.lipb.lipbid`));
+    raiSetUserId(bid, eids, 'adserver.org', utils.deepAccess(bid, `userId.tdid`))
+  }
+
+  return eids;
+}
+
+function raiSetUserId(bid, eids, source, value, userId) {
+  if (utils.isStr(value)) {
+    eids.push({
+      userId: value,
+      source
```

Comment:
should this be `source: source`? this is for your own ad request, so this may be the format you're expecting, but generally there's a key/value, not just a value, for each attribute.